transitions/core.py

```diff
@@ -359,16 +356,17 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
         self._transition_queue = deque()
         self.models = []
 
-        if initial is None:
-            self.add_states('initial')
+        if model is None and add_self:  # TODO(pbovbel) reconsider API for next major release
+            model = self
+
```

Comment:
Agreed re 0.5.x, in the meantime, can we keep add_self to maintain current behaviour/API but enable my use case?

Sometimes it doesn't make sense to provide an 'initial' state. In my case, I'm loading/unloading models all the time as they get cached from the db, meaning their 'initial' state is actually whatever the DB says.

I think the MachineError here makes sense as currently implemented (if no default 'initial' and no 'initial' in add_model) but i'm totally open to alternatives.